test/unit/server-statefulset.bats

```diff
@@ -11,6 +11,27 @@ load _helpers
   [ "${actual}" = "true" ]
 }
 
+
+@test "server/StatefulSet: affinity not set with server.affinity" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -x templates/server-statefulset.yaml  \
+      --set 'server.affinity=' \
+      . | tee /dev/stderr |
+      yq 'length > 0' | tee /dev/stderr)
```

Comment:
The `yq 'length > 0'` just checks that there _is_ output when we run `helm template` on the `server-statefulset.yaml`, so it doesn't give us much information about the `affinity` variable that we're trying to check.

In the case where `affinity=null`, we want to make sure that the `affinity:` section doesn't show up in the yaml file. Try this instead:
```suggestion
      yq '.spec.template.spec | any(contains("affinity"))' | tee /dev/stderr)
```

The condition needs to be updated to `false` with this change (I've created another suggestion for that).

Reply:
Okay, glad you found something that worked!